### DISCUSSION TITLE
Messages sent via API to WhatsApp with templates are sent correctly but do not appear in the inbox

### DISCUSSION THREAD

**Discussion by xtianfab** [chatwoot/chatwoot#10468] [Q&A]
Hello, messages sent via API to Whatsapp with templates are sent correctly but do not appear in the inbox of Chatwoot.

The conversation is created but the sent message does not appear.

![image](https://github.com/user-attachments/assets/5e1e29c4-3a0b-4011-8b67-2f4191ade885)

What could be the problem?

**Comment by sojan-official** [comment]
If  you are directly calling the whatsapp api the messages won't appear in Chatwoot. you would need to create messages in whatsapp by calling the Chatwoot api   and let chatwoot create this message internally as well as on whatsapp

**Reply by xtianfab** [reply]
Hello @sojan-official thanks for your answer.

I am using the chatwoot api to create the conversation. As I mentioned, the conversation is created in the chatwoot interface but the message does not appear. 

Apparently the problem is when I send messages to WhatsApp using templates. I tried adding messages to an existing conversation and the sent message appears. But when I send a template to the same conversation, the message does not appear.

Im using this url: https://cloud.qantris.com/api/{account_id}/accounts/1/conversations

payload = {
  "contact_id": "8",
  "inbox_id": "1",
  "message": {
    "template_params": {
      "name": "template_name",
      "category": "UTILITY",
      "language": "es",
      "processed_params": {
        "1": "param_1",
        "2": "param_2",
        "3": "param_3"
      }
    }
  }
}

Response 

{
  "meta": {
    "sender": {
      "additional_attributes": {
        "city": "",
        "country": "",
        "description": "",
        "company_name": "",
        "country_code": "",
        "social_profiles": {
          "github": "",
          "twitter": "",
          "facebook": "",
          "linkedin": "",
          "instagram": ""
        }
      },
      "availability_status": "offline",
      "email": "email",
      "id": 8,
      "name": "name",
      "phone_number": "+595xxxxxxxx",
      "identifier": null,
      "thumbnail": "",
      "custom_attributes": {},
      "last_activity_at": 1732120730,
      "created_at": 1731695322
    },
    "channel": "Channel::Whatsapp",
    "hmac_verified": false
  },
  "id": 80,
  "messages": [
    {
      "id": 176,
      "content": null,
      "account_id": 1,
      "inbox_id": 1,
      "conversation_id": 80,
      "message_type": 1,
      "created_at": 1732165411,
      "updated_at": "2024-11-21T05:03:31.173Z",
      "private": false,
      "status": "sent",
      "source_id": null,
      "content_type": "text",
      "content_attributes": {},
      "sender_type": "User",
      "sender_id": 1,
      "external_source_ids": {},
      "additional_attributes": {
        "template_params": {
          "name": "template_name",
          "category": "UTILITY",
          "language": "es",
          "processed_params": {
            "1": "param_1",
            "2": "param_2",
            "3": "param_3"
          }
        }
      },
      "processed_message_content": null,
      "sentiment": {},
      "conversation": {
        "assignee_id": null,
        "unread_count": 0,
        "last_activity_at": 1732165411,
        "contact_inbox": {
          "source_id": "595xxxxxxx"
        }
      },
      "sender": {
        "id": 1,
        "name": "name",
        "available_name": "available_name",
        "avatar_url": "",
        "type": "user",
        "availability_status": "offline",
        "thumbnail": ""
      }
    }
  ],
  "account_id": 1,
  "uuid": "cb19219d-5a9f-4ec6-8514-46143b9f6004",
  "additional_attributes": {},
  "agent_last_seen_at": 0,
  "assignee_last_seen_at": 0,
  "can_reply": false,
  "contact_last_seen_at": 0,
  "custom_attributes": {},
  "inbox_id": 1,
  "labels": [],
  "muted": false,
  "snoozed_until": null,
  "status": "open",
  "created_at": 1732165411,
  "timestamp": 1732165411,
  "first_reply_created_at": 1732165411,
  "unread_count": 0,
  "last_non_activity_message": {
    "id": 176,
    "content": null,
    "account_id": 1,
    "inbox_id": 1,
    "conversation_id": 80,
    "message_type": 1,
    "created_at": 1732165411,
    "updated_at": "2024-11-21T05:03:31.173Z",
    "private": false,
    "status": "sent",
    "source_id": null,
    "content_type": "text",
    "content_attributes": {},
    "sender_type": "User",
    "sender_id": 1,
    "external_source_ids": {},
    "additional_attributes": {
      "template_params": {
        "name": "template_name",
        "category": "UTILITY",
        "language": "es",
        "processed_params": {
          "1": "param_1",
          "2": "param_2",
          "3": "param_3"
        }
      }
    },
    "processed_message_content": null,
    "sentiment": {},
    "conversation": {
      "assignee_id": null,
      "unread_count": 0,
      "last_activity_at": 1732165411,
      "contact_inbox": {
        "source_id": "595xxxxxxxx"
      }
    },
    "sender": {
      "id": 1,
      "name": "name",
      "available_name": "available_name",
      "avatar_url": "",
      "type": "user",
      "availability_status": "offline",
      "thumbnail": ""
    }
  },
  "last_activity_at": 1732165411,
  "priority": null,
  "waiting_since": 0,
  "sla_policy_id": null
}
